setup_resources.sh

```diff
@@ -0,0 +1,157 @@
+ALL_SET="true"
+if [ -z ${AZURE_SUBSCRIPTION_ID} ]; then ALL_SET="false" && echo "AZURE_SUBSCRIPTION_ID is unset"; else echo "AZURE_SUBSCRIPTION_ID is set to '$AZURE_SUBSCRIPTION_ID'"; fi
+if [ -z ${LOCATION} ]; then ALL_SET="false" && echo "LOCATION is unset"; else echo "LOCATION is set to '$LOCATION'"; fi
+if [ -z ${CLUSTER_RESOURCEGROUP} ]; then ALL_SET="false" && echo "CLUSTER_RESOURCEGROUP is unset"; else echo "CLUSTER_RESOURCEGROUP is set to '$CLUSTER_RESOURCEGROUP'"; fi
+if [ -z ${CLUSTER_NAME} ]; then ALL_SET="false" && echo "CLUSTER_NAME is unset"; else echo "CLUSTER_NAME is set to '$CLUSTER_NAME'"; fi
+if [ -z ${CLUSTER_VNET} ]; then CLUSTER_VNET="aro-vnet2"; echo "CLUSTER_VNET is ${CLUSTER_VNET}"; fi
+if [ -z ${CLUSTER_MASTER_SUBNET} ]; then CLUSTER_MASTER_SUBNET="master-subnet"; echo "CLUSTER_MASTER_SUBNET is ${CLUSTER_MASTER_SUBNET}"; fi
+if [ -z ${CLUSTER_WORKER_SUBNET} ]; then CLUSTER_WORKER_SUBNET="worker-subnet"; echo "CLUSTER_WORKER_SUBNET is ${CLUSTER_WORKER_SUBNET}"; fi
+if [ -z ${OPENSHIFT_VERSION} ]; then ALL_SET="false" && echo "OPENSHIFT_VERSION is unset"; else echo "OPENSHIFT_VERSION is set to '4.12.25'"; fi
+
+if [[ "${ALL_SET}" != "true" ]]; then exit 1; fi
+
+# Check Azure CLI version
+echo "Checking Azure CLI version..."
+az_version=$(az --version | grep 'azure-cli' | awk '{print $2}')
+required_version="2.30.0"
+if  [ "$(printf '%s\n' "$required_version" "$az_version" | sort -V | head -n1)" = "$required_version" ]; then
+    echo "Azure CLI version is compatible"
+else
+    echo "Azure CLI version must be $required_version or later. Please upgrade."
+    exit 1
+fi
+
+# Set the subscription
+echo "Setting the subscription..."
+az account set --subscription $AZURE_SUBSCRIPTION_ID
+
+# Register the subscription directly
+echo "Registering the subscription directly..."
+curl -k -X PUT \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "state": "Registered",
+    "properties": {
+        "tenantId": "'"$AZURE_TENANT_ID"'",
+        "registeredFeatures": [
+            {
+                "name": "Microsoft.RedHatOpenShift/RedHatEngineering",
+                "state": "Registered"
+            }
+        ]
+    }
+}' "https://localhost:8443/subscriptions/$AZURE_SUBSCRIPTION_ID?api-version=2.0"
+
+# Function to add supported OpenShift version
+add_openshift_version() {
+  local version=$1
+  local openshift_pullspec=$2
+  local installer_pullspec=$3
+
+  echo "Adding OpenShift version $version..."
+  curl -k -X PUT "https://localhost:8443/admin/versions" --header "Content-Type: application/json" -d '{
+    "properties": {
+      "version": "'"$version"'",
+      "enabled": true,
+      "openShiftPullspec": "'"$openshift_pullspec"'",
+      "installerPullspec": "'"$installer_pullspec"'"
+    }
+  }'
+}
+
+# Add the required OpenShift version
+add_openshift_version "4.12.25" "quay.io/openshift-release-dev/ocp-release@sha256:<sha256>" "<name>.azurecr.io/installer:release-4.x"
```

Comment:
I think we should plug environment variables into the correct places here to set the OCP and installer image pullspecs. As it is right now, the script will put these strings into Cosmos DB 
as-is and then cluster creation will fail because these aren't valid pullspecs. This is easy to miss and would be frustrating and time-consuming to deal with after making the mistake.